### PR TITLE
Pass cilium config to use it as a kube-proxy replacement

### PR DESCRIPTION
**Pull request by fiunchinho**
Towards https://github.com/giantswarm/roadmap/issues/1455

### Checklist

- [X] Update changelog in `CHANGELOG.md`.
- [ ] Make sure `values.yaml` is valid.

### CODE REVIEW

File: helm/default-apps-aws/values.yaml [hunk 1]
```diff
@@ -7,6 +7,12 @@ userConfig:
       values: |
         ipam:
           mode: kubernetes
+        k8sServiceHost: api.{{ .Values.baseDomain }}
+        k8sServicePort: 6443
```
Comment (mnitchev) [inline helm/default-apps-aws/values.yaml:11]: Isn't it 443 if it's using the LB?

Comment (fiunchinho) [reply helm/default-apps-aws/values.yaml:11]: It's not https://github.com/giantswarm/giantswarm/issues/22518. I actually tested it with 6443.

File: helm/default-apps-aws/values.yaml [hunk 2]
```diff
@@ -7,6 +7,12 @@ userConfig:
...
+        kubeProxyReplacement: strict
+        hubble:
+          relay:
+            enabled: true
```
Comment (whites11) [inline helm/default-apps-aws/values.yaml:15]: Not a blocker, but why don't we enable this by default, in the app?
We should IMHO strive to make cilium configuration as equal as possible between providers, and having custom settings like this one in provider-specific repos me don't like

**Review by whites11 — APPROVED** [review]

File: helm/default-apps-aws/values.yaml (hunk 2 above)
Comment (fiunchinho) [reply helm/default-apps-aws/values.yaml:15]: I agree with that. Do you know if that could break something in vintage? Currently it's like that for CAPG and CAPA. If you say it's fine on vintage, I can move that config to the app itself and remove it from here.

Comment (whites11) [reply helm/default-apps-aws/values.yaml:15]: In WCs cilium is still a dream.
In MCs I will test it with next bunch of releases I guess.
Safe to change the default if you ask me. 
Maybe worth asking rocket as well

**Pull request merged**